Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -1074,7 +1118,7 @@ void AsyncScrollingCoordinator::setSynchronousScrollingReasons(ScrollingNodeID n
 
 OptionSet<SynchronousScrollingReason> AsyncScrollingCoordinator::synchronousScrollingReasons(ScrollingNodeID nodeID) const
 {
-    RefPtr node = dynamicDowncast<ScrollingStateScrollingNode>(m_scrollingStateTree->stateNodeForID(nodeID));
+    RefPtr node = dynamicDowncast<ScrollingStateScrollingNode>(stateNodeForID(nodeID));
```

Comment:
This patch would be much smaller if you did all the `stateNodeForID()` changes first.

Reply:
That's how it started but it wasn't testable without the rest.  Now that we know what it takes to get the rest, it could be fine to split it up.  No preference from me.

Reply:
Planning on splitting this up now.